evaluation/depth_metrics.py

```diff
@@ -0,0 +1,101 @@
+# coding=utf-8
+# Copyright 2021 The Deeplab2 Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Implementation of the metrics for monocular depth estimation.
+
+The metrics for monocular depth estimation includes scale invariant logarithmic
+error (SILog), relative squared error (SqErrorRel), relative absolute error
+(AbsErrorRel), and depth inlier metric (DepthInlier) at 1.25.
+
+Please see the paper for details:
+"Sparsity Invariant CNNs", Jonas Uhrig, Nick Schneider, Lukas Schneider, Uwe
+Franke, Thomas Brox, and Andreas Geiger. 3DV 2017.
+"""
+
+from typing import Dict, Text, Any
+import tensorflow as tf
+
+
+class DepthMetrics(tf.keras.metrics.Metric):
+  """Metric class for the metrics for monocular depth estimation.
+
+  The metrics for monocular depth estimation includes scale invariant
+  logarithmic error (SILog), relative squared error (SqErrorRel), relative
+  absolute error (AbsErrorRel), and depth inlier metric (DepthInlier) at 1.25.
+
+  "Sparsity Invariant CNNs", Jonas Uhrig, Nick Schneider, Lukas Schneider, Uwe
```

Comment:
Also add this paper

Depth Map Prediction from a Single Image
using a Multi-Scale Deep Network